homeassistant/util/unit_system.py

```diff
@@ -213,7 +216,13 @@ def _deprecated_unit_system(value: str) -> str:
     """Convert deprecated unit system."""
 
     if value == _CONF_UNIT_SYSTEM_IMPERIAL:
-        # need to add warning in 2023.1
+        # Deprecated in 2024.12, to raise error in 2025.12
+        _LOGGER.warning(
+            "Your Home Assistant configuration is using the imperial unit system"
+            " which is deprecated, please replace imperial with us_customary"
+            " in your Home Assistant configuration and restart Home Assistant"
```

Comment:
Where is this used? would it make sense to create an issue in the issue registry?

Reply:
It's a bit tricky as the validation happens in the schema (which has no access to `hass`) and then it's migrated in store. So I need to have a second look and perhaps move the check out of the schema validation completely.